firebase/FirestoreCtrl.ts

```diff
@@ -180,4 +180,28 @@ export default class FirestoreCtrl {
       throw error;
     }
   }
+
+  /**
+   * Retrieves all challenges from Firestore. WARNING: This can be slow and expensive.
+   *
+   * @returns A promise that resolves to an array of all challenges.
+   */
+  async getAllChallenges(): Promise<DBChallenge[]> {
+    try {
+      const challengesRef = collection(firestore, "challenges");
+      const querySnapshot = await getDocs(challengesRef);
+      const challenges = querySnapshot.docs.map((doc) => {
+        const data = doc.data();
+        console.log("+ Challenge data retrieved:", data);
```

Comment:
Same here, if the number of challenges is big, it may not be very wise to log something everytime we retrieve a challenge, but for now I think it's okay.

Reply:
This was used for debug purposes and is not needed this precisely anymore. I'll change that to a single log at the end.